### PR TITLE
feat: local drive support on AHS simulator

### PR DESCRIPTION
**Pull request by mudit2812**
*Description of changes:*
* Added helpers to `ahs_translation.py` for converting pulses into `ShiftingField`.
* Updated `BraketAhsDevice.generate_samples` to return a numpy array so that common return types such as `qml.probs`, `qml.expval`, `qml.var` can be used.
* Updated `setup.py` to check if the system is an ARM Mac to update the installed Tensorflow version correctly.

*Testing done:*
* Unit tests for local drive helpers
* Integration tests to validate that qnodes work with various return types.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -62,8 +68,8 @@
             "sphinx-rtd-theme",
             "sphinxcontrib-apidoc",
             "tox",
-            "tensorflow>=2.6.0",
```
Comment (speller26) [inline setup.py:65]: Why remove TensorFlow?

File: setup.py [hunk 2]
```diff
@@ -12,13 +12,19 @@
...
 with open("README.rst", "r") as fh:
     long_description = fh.read()
 
 with open("src/braket/pennylane_plugin/_version.py") as f:
     version = f.readlines()[-1].split()[-1].strip("\"'")
 
+if platform.system() == "Darwin" and platform.machine() == "arm64":
+    TF_VERSION = "tensorflow-macos>=2.6.0"
+else:
+    TF_VERSION = "tensorflow>=2.6.0"
```
Comment (speller26) [inline setup.py:26]: Can this be updated to 2.12?

**Comment by mudit2812** [comment]
I'm having some linting errors. Some of the docstrings in `ahs_device.py` have links to the Pennylane docs for important functions such as `rydberg_drive`, `rydberg_interaction`, etc. The links are too long to fit within the 100 character limit, so I'm updating `tox.ini` to ingore the warnings in that file.

File: src/braket/pennylane_plugin/ahs_translation.py [hunk 3]
```diff
@@ -195,3 +200,134 @@ def _get_sample_times(time_interval: ArrayLike):
...
+        if not np.allclose(time_slice, new_time_slice):
+            raise ValueError(
+                "Local detunings don't match. Make sure that all local detunings match "
+                "in shape and only differ in magnitude."
+            )
+
+    return max_detuning, Pattern(pattern)
+
+
+def translate_pulses_to_shifting_field(detunings, time_points):
```
Comment (speller26) [inline src/braket/pennylane_plugin/ahs_translation.py:317]: Needs type hints
```suggestion
def translate_pulses_to_shifting_field(
    detunings: List[Union[float, callable]], time_points: np.array
):
```

File: src/braket/pennylane_plugin/ahs_translation.py [hunk 4]
```diff
@@ -195,3 +200,134 @@ def _get_sample_times(time_interval: ArrayLike):
...
+    """Uses the overall detuning and pattern to create a ``ShiftingField`` object from
+    AWS Braket.
+
+    Args:
+        detunings (List[Union[float, callable]]): Local detuning per wire
+        time_points (array[Number, Number]]): Array of sampled time steps
+
+    Returns:
+        ShiftingField: the object representing the local drive for the
+        ``AnalogueHamiltonianSimulation`` object
```
Comment (speller26) [inline src/braket/pennylane_plugin/ahs_translation.py:327]:
```suggestion
        ``AnalogHamiltonianSimulation`` object
```

File: src/braket/pennylane_plugin/ahs_device.py [hunk 5]
```diff
@@ -443,8 +457,108 @@ def settings(self):
         # C6 for the Rubidium transition used by the simulator, converted to MHz x um^6
         return {"interaction_coeff": 862620}
 
+    def _ahs_program_from_evolution(self, evolution):
+        """Create AHS program for simulation from a ParametrizedEvolution
+
+        Args:
+            evolution (ParametrizedEvolution): the PennyLane operator describing the pulse
+                to be converted into an Analogue Hamiltonian Simulation program
```
Comment (speller26) [inline src/braket/pennylane_plugin/ahs_device.py:465]:
```suggestion
                to be converted into an analog Hamiltonian simulation program
```

**Comment by speller26** [comment]
Looks like there are multiple test failures too

File: src/braket/pennylane_plugin/ahs_device.py [hunk 6]
```diff
@@ -443,8 +457,108 @@ def settings(self):
...
+        H = translate_pulse_to_driving_field(self._pulses[self._global_pulse_idx], time_points)
+
+        # Create local detunings
+        local_pulses = self._pulses.copy()
+        local_pulses.pop(self._global_pulse_idx)
+
+        local_detunings = _create_valid_local_detunings(local_pulses, self.wires)
+        if local_detunings is not None:
+            shift = translate_pulses_to_shifting_field(local_detunings, time_points)
+            H = H + shift
```
[the remark this answers is not in the document]
Comment (mudit2812) [reply src/braket/pennylane_plugin/ahs_device.py:487]: I tried doing this and it raises an error.

**Comment by speller26** [comment]
@mudit2812 Can I close this PR?

**Comment by mudit2812** [comment]
@speller26 Yep, the branch that was merged over the weekend had the changes from this branch merged into it as far as I'm aware.

**Comment by trbromley** [comment]
This PR is to be closed.

**Pull request closed without merging**